src/module.ts

```diff
@@ -137,6 +141,8 @@ export default defineNuxtModule<ModuleOptions>({
     options.api.methods = moduleOptions.api.methods.length > 0 ? moduleOptions.api.methods : ['patch', 'delete', 'get', 'post']
     nuxt.options.runtimeConfig.session = defu(nuxt.options.runtimeConfig.session, options)
     nuxt.options.runtimeConfig.public = defu(nuxt.options.runtimeConfig.public, { session: { api: options.api } })
+    const nitroStorageOptions = defu(nuxt.options.nitro.storage, { [options.session.storePrefix]: options.session.storageOptions })
+    nuxt.options.nitro.storage = nitroStorageOptions
```

Comment:
I think the attempt will work, but instead of relying on nitro and overwriting it's storage options, let's continue to use `unstorage` directly: This way the user can still configure their nitro-storage that they are using in their app, potentially for things additional to session storage and can use a different set of configuration for their session-management.

E.g. they could want to use redis-storage for their sessions while staying memory based for their application.

Reply:
This shouldn't be overwriting nitro's storage options, only adding to them (unless there's also a `sessions` storage in the nitro settings. Then it'd get overwritten...). 
So if you had a configuration like this
```js
export default defineNuxtConfig({
  modules: [NuxtSession],
  nitro: {
    storage: {
      db: {
        driver: 'fs',
        base: './tmp'
      }
    }
  },
  session: {
    session: {
      storageOptions: {
        driver: 'redis'
      }
    }
  }
})
```
You could access Nitro's built-in "cache" storage thats in memory by default with:
```js
  const storage = useStorage()
  await storage.setItem('cache:myItem', 'myitem')
  await storage.getItem('cache:myItem')
```
the "db" storage with the `fs` driver:
```js
  const storage = useStorage()
  await storage.setItem('db:myItem', 'myitem')
  await storage.getItem('db:myItem')
```
and then this module's "sessions" storage using the `redis` driver:
```js
  const storage = useStorage()
  await storage.setItem('sessions:myItem', 'myitem')
  await storage.getItem('sessions:myItem')
```

But I do like the idea of not using Nitro's configuration since that'd avoid any accidental overriding. We could probably do something similar to what it does [here](https://github.com/unjs/nitro/blob/484261aa747b544a3ff7a116e1ac1116d2856b2c/src/storage.ts#L4)
and create a `useSessionStorage` function similar to `useStorage` just only for this module. 

I'll see if I can work on this more today

Reply:
Gotcha - you're right about the not overwriting + I like the proposal you've given.

About the current implementation I do like that it's making use of nitro to let it setup the storage. Only problem being that then the import in the module of `useStorage` does not work out.

Have you had the time to further this or the proposed implementation? (: I would like to squash this bug soon, as it's important for the library. If you do not have the time to further this that's totally fine, but then at some point I would take over. 

Thanks & Cheers 🍻

Reply:
I haven't been able to work on it yet, and probably won't be able to today. Its totally fine if you wanna take over though!

Reply:
Do you have an ETA at what point you want to continue your work here? (: I'll be on vacation starting end of the week and would want to merge this (or a similar) PR before then.

Reply:
I think I could get it in by the end of this week. I'll shoot for Wednesday or Thursday

Reply:
Alright, then let's try that & sounds great! Afterwards I'll do the release (: 

FYI: I clearer the de-/serialization issue and added some docs that this is intentional, see https://github.com/nuxt/framework/issues/8419 for more context.